google-cloud-spanner/src/main/java/com/google/cloud/spanner/DelayedTransactionManager.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spanner;
+
+import com.google.api.core.ApiFuture;
+import com.google.cloud.Timestamp;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Represents a {@link TransactionManager} using a multiplexed session that is not yet ready. The
+ * execution will be delayed until the multiplexed session has been created and is ready. This class
+ * is only used during the startup of the client and the multiplexed session has not yet been
+ * created.
+ */
+class DelayedTransactionManager implements TransactionManager {
+
+  private final ApiFuture<TransactionManager> transactionManagerFuture;
+
+  DelayedTransactionManager(ApiFuture<TransactionManager> transactionManagerFuture) {
+    this.transactionManagerFuture = transactionManagerFuture;
+  }
+
+  TransactionManager getTransactionManager() {
+    try {
+      return this.transactionManagerFuture.get();
+    } catch (ExecutionException executionException) {
+      // Propagate the underlying exception as a RuntimeException (SpannerException is also a
+      // RuntimeException).
+      if (executionException.getCause() instanceof RuntimeException) {
+        throw (RuntimeException) executionException.getCause();
+      }
+      throw SpannerExceptionFactory.asSpannerException(executionException.getCause());
```

Comment:
I have not clearly understood about this comment. Can you please elaborate?

Reply:
Hmm... I actually read the code block wrong, so my comment did indeed not make any sense.

But: Could we add a util method to `SpannerExceptionFactory` that contains the code block in this catch block and use that in all of these catch blocks, instead of repeating it? So like this:

```java
RuntimeException causeAsRuntimeException(ExecutionException executionException) {
      if (executionException.getCause() instanceof RuntimeException) {
        return (RuntimeException) executionException.getCause();
      }
      return asSpannerException(executionException.getCause());
}
```

And then simplify the catch blocks to this:

```java
    } catch (ExecutionException executionException) {
      throw SpannerExceptionFactory.causeAsRuntimeException(executionException);
  }
```

Reply:
Thanks, Done